runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -9621,6 +9623,13 @@ runMethodHandle: {
 	VMStructHasBeenUpdated(REGISTER_ARGS);
 	goto throwCurrentException;
 
+nullPointerWithExtendedMsg:
+	updateVMStruct(REGISTER_ARGS);
+	prepareForExceptionThrow(_currentThread);
+	setCurrentExceptionUTF(_currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, getExtendedNPEMessage(_currentThread, *_pc, *(U_16*)(_pc + 1), J9_CP_FROM_METHOD(_literals)));
```

Comment:
Might be better to simply pass `_currentThread` to the message helper and let it fetch the data that's required.

Also need to verify that every path here has not incremented the pc (I suspect this is already true).

Reply:
`prepareForExceptionThrow(_currentThread)` could update `_currentThread->_pc`, hence `getExtendedNPEMessage()` can't retrieve current `_pc` value via `_currentThread`.

Reply:
@gacholio @DanHeidinga this is ready for review.